ludwig/models/predictor.py

```diff
@@ -366,7 +376,17 @@ def batch_evaluation(self, dataset, collect_predictions=False, collect_logits=Fa
                 progress_bar = LudwigProgressBar(self.report_tqdm_to_ray, progress_bar_config, self.is_coordinator())
 
                 predictions = defaultdict(list)
+                eval_steps = (
+                    self.dist_model.config_obj.trainer.eval_steps
+                    if hasattr(self.dist_model, "config_obj")
+                    and hasattr(self.dist_model.config_obj.trainer, "eval_steps")
+                    else None
+                )
+                eval_steps_counter = 0
                 while not batcher.last_batch():
+                    if eval_steps and eval_steps_counter >= eval_steps:
+                        logger.info(f"Reached evaluation step {eval_steps}. Ending evaluation.")
```

Comment:
@Infernaught can you write a test that confirms the first `n` samples of the evaluation are what are evaluated against per epoch?

Reply:
Done!